ipatests/test_integration/test_installation_client.py

```diff
@@ -70,3 +75,115 @@ def test_client_install_with_ssh_trust_dns(self):
+        3H      ;Refresh
+        15M     ;Retry
+        1W      ;Expire
+        1D      ;Minimum 1D
+        )
+        @      IN  NS      {fqdn}.
+        ns1 IN  A       {bindserverip}
+        _kerberos.{domain}. IN TXT {zoneupper}
+        {fqdn}.    IN  A       {bindserverip}
+        ipa-ca.{domain}.        IN  A       {bindserverip}
+        _kerberos-master._tcp.{domain}. IN SRV 0 100 88 {fqdn}.
+        _kerberos-master._udp.{domain}. IN SRV 0 100 88 {fqdn}.
+        _kerberos._tcp.{domain}. 	IN SRV 0 100 88 {fqdn}.
+        _kerberos._udp.{domain}. 	IN SRV 0 100 88 {fqdn}.
+        _kpasswd._tcp.{domain}. 	IN SRV 0 100 464 {fqdn}.
+        _kpasswd._udp.{domain}. 	IN SRV 0 100 464 {fqdn}.
+        _ldap._tcp.{domain}. 		IN SRV 0 100 389 {fqdn}.
+        """).format(
+            fqdn=bindserver.hostname,
+            domain=bindserver.domain.name,
+            bindserverip=bindserver.ip,
+            zoneupper=bindserver.domain.name.upper()
+        )
+        bindserverdb = "/var/named/{0}.db".format(bindserver.domain.name)
+        bindserver.put_file_contents(bindserverdb, add_records)
+        bindserver.run_command(['systemctl', 'start', 'named'])
+        Firewall(bindserver).enable_services(["dns"])
+        yield
+        named_conf_backup.restore()
+        bindserver.run_command(['rm', '-rf', bindserverdb])
+
+    def test_client_nsupdate(self, setup_bindserver):
+        """Test secure nsupdate failed, then try unsecure nsupdate..
+
+        Test to verify when bind is configured with dynamic update policy,
+        and during client-install 'nsupdate -g' fails then it should run with
+        second call using unauthenticated nsupdate.
+
+        Related : https://pagure.io/freeipa/issue/8402
+        """
+        # with pre-configured bind server, install ipa-server without dns.
+        tasks.install_master(self.master, setup_dns=False)
+        self.client.resolver.backup()
+        self.client.resolver.setup_resolver(
+            self.master.ip, self.master.domain.name)
+        try:
+            self.client.run_command(['ipa-client-install', '-U',
+                                     '--domain', self.client.domain.name,
+                                     '--realm', self.client.domain.realm,
+                                     '-p', self.client.config.admin_name,
+                                     '-w', self.client.config.admin_password,
+                                     '--server', self.master.hostname])
+            # call unauthenticated nsupdate if GSS-TSIG nsupdate failed.
+            str1 = "nsupdate (GSS-TSIG) failed"
+            str2 = "'/usr/bin/nsupdate', '/etc/ipa/.dns_update.txt'"
+            client_log = self.client.get_file_contents(
+                paths.IPACLIENT_INSTALL_LOG, encoding='utf-8'
+            )
+            assert str1 in client_log and str2 in client_log
+            dig_after = self.client.run_command(
+                ['dig', self.client.hostname])
```

Comment:
The test output shows that the client's systemd-resolved resolver is used for this command:
```
;; SERVER: 127.0.0.53#53(127.0.0.53)
```
The DNS server needs to be specified (dig @self.master.ip self.client.hostname), otherwise the local systemd-resolved process is used and resolves the client name through the content of /etc/hosts.